### PR TITLE
Adaptive time intervals for embedded dictionaries loading [#METR-22050].

### PR DESCRIPTION
**Pull request by ludv1x**

### CODE REVIEW

File: dbms/include/DB/Interpreters/Dictionaries.h [hunk 1]
```diff
@@ -116,28 +118,46 @@ class Dictionaries
...
 		}
 	}
 
 public:
-	/// Справочники будут обновляться в отдельном потоке, каждые reload_period секунд.
+	/// Every reload_period seconds directories are updated inside a separate thread.
 	Dictionaries(const bool throw_on_error, const int reload_period_)
-		: reload_period(reload_period_), log(&Logger::get("Dictionaries"))
+		: reload_period(reload_period_),
+		cur_reload_period(1),
```
Comment (alexey-milovidov) [inline dbms/include/DB/Interpreters/Dictionaries.h]: Better to initialize in place (at member declaration).

File: dbms/include/DB/Interpreters/Dictionaries.h [hunk 2]
```diff
@@ -116,28 +118,46 @@ class Dictionaries
...
+			if (reloadImpl())
+			{
+				/// Sucess
+				cur_reload_period = reload_period;
+				is_fast_start_stage = false;
+			}
+
+			if (is_fast_start_stage)
+			{
+				cur_reload_period = std::min(reload_period, 2*cur_reload_period); /// exponentially increase delay
```
Comment (alexey-milovidov) [inline dbms/include/DB/Interpreters/Dictionaries.h]: Need whitespaces.

File: dbms/include/DB/Interpreters/Dictionaries.h [hunk 3]
```diff
@@ -116,28 +118,46 @@ class Dictionaries
...
 		while (true)
 		{
-			if (destroy.tryWait(reload_period * 1000))
+			if (destroy.tryWait(cur_reload_period * 1000))
 				return;
 
-			reloadImpl();
+			if (reloadImpl())
+			{
+				/// Sucess
```
Comment (alexey-milovidov) [inline dbms/include/DB/Interpreters/Dictionaries.h]: Typo.

**Comment by ludv1x** [comment]
@alexey-milovidov  Updated

**Pull request merged**